dig.go

```diff
@@ -210,14 +259,69 @@ func (c *Container) get(t reflect.Type) (reflect.Value, error) {
 
 	for _, con := range constructed {
 		ct := con.Type()
-		if ct == _errType {
-			continue
-		}
-		c.cache[ct] = con
+		c.set(ct, con)
```

Comment:
Can `set` take just `con` and handle getting the type internally? Prevents unintentional mismatches between the passed value and type.